src/test/scala/zio/interop/TwitterSpec.scala

```diff
@@ -3,80 +3,75 @@ package zio.interop
+        suite("Task.fromTwitterFuture")(
+          testM("return failing `Task` if future failed.") {
+            val error                              = new Exception
+            val future                             = Task(Future.exception[Int](error))
+            val task                               = Task.fromTwitterFuture(future).unit
+            val assertion: Assertion[Either[Throwable, Unit]] = equalTo(Left(error))
+            assertM(task.either, assertion)
+          },
+          testM("return successful `Task` if future succeeded.") {
+            val value  = 10
+            val future = Task(Future.value(value))
+            val task   = Task.fromTwitterFuture(future).option
+            val assertion: Assertion[Option[Int]] = equalTo(Some(value))
+            assertM(task, assertion)
+          },
+          testM("ensure future is interrupted together with task.") {
+            val value = new AtomicInteger(0)
+            val promise = new Promise[Unit] with Promise.InterruptHandler {
+              override protected def onInterrupt(t: Throwable): Unit = setException(t)
+            }
+            val future = Task(promise.flatMap(_ => Future(value.incrementAndGet())))
+            val task = 
+              (for {
+                fiber <- Task.fromTwitterFuture(future).fork
+                _     <- fiber.interrupt
+                _     <- Task.effect(promise.setDone())
+                a     <- fiber.await
+              } yield a).fold(_ => false, exit => exit.toEither.isLeft)
+              
+            task.map { b =>
+              assert(b, isTrue) && assert(value.get(), equalTo(0))
+            }
+          }
+        )
+      ,
+      suite("Runtime.unsafeRunToTwitterFuture")(
+        test("return successful `Future` if Task evaluation succeeded.") {
+          assert(Await.result(Runtime((), DefaultTestRunner.platform).unsafeRunToTwitterFuture(Task.succeed(2))), equalTo(2))
+        },
+        test("return failed `Future` if Task evaluation failed.") {
+          val e    = new Throwable
+          val task = Task.fail(e).unit
+          val result = Try(Await.result(Runtime((), DefaultTestRunner.platform).unsafeRunToTwitterFuture(task))).toEither
+          assert(result, isLeft(equalTo(e)))
+        },
+        test("ensure Task evaluation is interrupted together with Future.") {
+          val value = new AtomicInteger(0)
+          val ex = new Throwable
+          val task: ZIO[Any, Throwable, Future[Int]] = for {
+            promise <- zio.Promise.make[Throwable, Int]
+            t       = promise.await.flatMap( _ =>               Task.effectTotal(value.incrementAndGet())       )
+            future  = Runtime((), DefaultTestRunner.platform).unsafeRunToTwitterFuture(t)
+            _       = future.raise(ex)
+            _       <- promise.succeed(1)
+          } yield future
+
+          val result = Try(Await.result(Runtime((), DefaultTestRunner.platform).unsafeRun(task))).toEither
```

Comment:
This line seems to create an exception.

```
[error] java.lang.InterruptedException
[error] 	at zio.Cause.$anonfun$squashWith$1(Cause.scala:246)
[error] 	at scala.Option.orElse(Option.scala:306)
[error] 	at zio.Cause.squashWith(Cause.scala:246)
[error] 	at zio.Cause.squashWith$(Cause.scala:244)
[error] 	at zio.Cause$Traced.squashWith(Cause.scala:322)
[error] 	at zio.Cause.squash(Cause.scala:238)
[error] 	at zio.Cause.squash$(Cause.scala:237)
[error] 	at zio.Cause$Traced.squash(Cause.scala:322)
[error] 	at zio.interop.twitter.package$RuntimeOps$.failure$1(twitter.scala:41)
[error] 	at zio.interop.twitter.package$RuntimeOps$.$anonfun$unsafeRunToTwitterFuture$3(twitter.scala:51)
[error] 	at zio.interop.twitter.package$RuntimeOps$.$anonfun$unsafeRunToTwitterFuture$3$adapted(twitter.scala:51)
[error] 	at zio.Exit.fold(Exit.scala:89)
[error] 	at zio.Exit.fold$(Exit.scala:86)
[error] 	at zio.Exit$Failure.fold(Exit.scala:205)
[error] 	at zio.interop.twitter.package$RuntimeOps$.$anonfun$unsafeRunToTwitterFuture$2(twitter.scala:51)
[error] 	at zio.interop.twitter.package$RuntimeOps$.$anonfun$unsafeRunToTwitterFuture$2$adapted(twitter.scala:51)
[error] 	at zio.internal.FiberContext.$anonfun$runAsync$1$adapted(FiberContext.scala:142)
[error] 	at zio.internal.FiberContext.$anonfun$notifyObservers$2(FiberContext.scala:811)
[error] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[error] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[error] 	at java.lang.Thread.run(Thread.java:748)```

I expected the Try(...) to catch that. Maybe someone else sees what I'm missing here?

Reply:
That's because `Try` only catches `NonFatal` exceptions and `InterruptedException` is considered `Fatal`.

Reply:
You can use `Task(...)` and `assertM` instead.

Reply:
Oh, didn't know that it wasn't considered a NonFatal exception :) I'll make the necessary adjustments.